File: sdk/julia/src/will_expired.jl
# WillExpired: expiring services and domains management
struct WillExpiredResource
    http::HttpClient
end

# GET /api/willexpired - List expiring services and domains
function list_expiring(r::WillExpiredResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/willexpired"; params = params)
end

# GET /api/willexpired/summary - Expiring summary
function get_summary(r::WillExpiredResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/willexpired/summary"; params = params)
end

# GET /api/willexpired/config - Module configuration
function get_config(r::WillExpiredResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/willexpired/config"; params = params)
end

# GET /api/willexpired/invoices - Open renewal invoices
function list_open_invoices(r::WillExpiredResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/willexpired/invoices"; params = params)
end

# GET /api/willexpired/requests - Renewal request log
function list_requests(r::WillExpiredResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/willexpired/requests"; params = params)
end

# GET /api/willexpired/export - Export all expiring items
function export_all(r::WillExpiredResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/willexpired/export"; params = params)
end

# GET /api/willexpired/{type}/{id} - Get item detail
function get_item(r::WillExpiredResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/willexpired/{type}/{id}"; params = params)
end

# POST /api/willexpired/{type}/{id}/renew - Renew item
function renew(r::WillExpiredResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/willexpired/{type}/{id}/renew"; params = params)
end

# GET /api/willexpired/{type}/{id}/autorenew - Get auto-renew status
function get_autorenew(r::WillExpiredResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/willexpired/{type}/{id}/autorenew"; params = params)
end

# PUT /api/willexpired/{type}/{id}/autorenew - Set auto-renew
function set_autorenew(r::WillExpiredResource; kwargs...)
    params = Dict(kwargs)
    http_put(r.http, "/api/willexpired/{type}/{id}/autorenew"; params = params)
end

File: sdk/dart/lib/src/will_expired.dart
import 'http_client.dart';

/// WillExpired: expiring services and domains management
class WillExpiredResource {
  final HttpClient httpClient;
  WillExpiredResource(this.httpClient);

  /// GET /api/willexpired - List expiring services and domains
  Future<Map<String, dynamic>> listExpiring([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/willexpired', params);
  }

  /// GET /api/willexpired/summary - Expiring summary
  Future<Map<String, dynamic>> getSummary([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/willexpired/summary', params);
  }

  /// GET /api/willexpired/config - Module configuration
  Future<Map<String, dynamic>> getConfig([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/willexpired/config', params);
  }

  /// GET /api/willexpired/invoices - Open renewal invoices
  Future<Map<String, dynamic>> listOpenInvoices([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/willexpired/invoices', params);
  }

  /// GET /api/willexpired/requests - Renewal request log
  Future<Map<String, dynamic>> listRequests([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/willexpired/requests', params);
  }

  /// GET /api/willexpired/export - Export all expiring items
  Future<Map<String, dynamic>> exportAll([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/willexpired/export', params);
  }

  /// GET /api/willexpired/{type}/{id} - Get item detail
  Future<Map<String, dynamic>> getItem([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/willexpired/{type}/{id}', params);
  }

  /// POST /api/willexpired/{type}/{id}/renew - Renew item
  Future<Map<String, dynamic>> renew([Map<String, dynamic>? params]) async {
    return httpClient.post('/api/willexpired/{type}/{id}/renew', params);
  }

  /// GET /api/willexpired/{type}/{id}/autorenew - Get auto-renew status
  Future<Map<String, dynamic>> getAutorenew([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/willexpired/{type}/{id}/autorenew', params);
  }

  /// PUT /api/willexpired/{type}/{id}/autorenew - Set auto-renew
  Future<Map<String, dynamic>> setAutorenew([Map<String, dynamic>? params]) async {
    return httpClient.put('/api/willexpired/{type}/{id}/autorenew', params);
  }

}
